### PR TITLE
chore: replace external eslint action, continue on failure

### PR DESCRIPTION
**Pull request by JKRhb**
This PR replaces the external Github Action for `eslint` by first installing it using `npm ci` and then running `npm run lint`. This removes the action as one additional dependency and also replicates the local environment for linting while still keeping the annotations made in the "Files changed" tab.

Furthermore, this PR lets the linting workflow pass even if errors occur, making it easier to see if there is an actual failing test. Once all linting issues have been resolved, this addition can be removed again.

### CODE REVIEW

**Comment by danielpeintner** [comment]
General question: I wonder whether we want to have a green light in case of linting errors?

At the moment it is somewhat annoying to see the red x.
Having said that, once everything is in place I guess I would like to see the red x even in case for linting. I mean I can still check which task failed. Without any notification I assume it easily happens to merge PRs that should be first properly linted.

Any thoughts?

**Comment by JKRhb** [comment]
> General question: I wonder whether we want to have a green light in case of linting errors?
> 
> At the moment it is somewhat annoying to see the red x.
> Having said that, once everything is in place I guess I would like to see the red x even in case for linting. I mean I can still check which task failed. Without any notification I assume it easily happens to merge PRs that should be first properly linted.
> 
> Any thoughts?

This is exactly what I was thinking :) Therefore this PR gives a green light for the linting action even if it fails but still displays the errors on the "Files changed" tab. When everything is fixed we can let the CI fail again if any issues are detected but for now this should make the process of opening PRs less annoying.

**Comment by relu91** [comment]
Ok for me, but please open an issue as a reminder to re-enable linting when we have finished the fix the linting process. Having a red flag for linting error is really the point of having linting in the codebase :)

**Review by relu91 — APPROVED** [review]
Just to be clear we can go on :) @danielpeintner I let you have the final decision.

**Comment by JKRhb** [comment]
> Ok for me, but please open an issue as a reminder to re-enable linting when we have finished the fix the linting process. Having a red flag for linting error is really the point of having linting in the codebase :)

Done :) I opened #516 as a reminder :)

**Review by danielpeintner — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
